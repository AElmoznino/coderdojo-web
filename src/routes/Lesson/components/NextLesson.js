// @flow

import React from 'react'
import styled from 'styled-components'
import GridSection from 'components/Grid/GridSection'
import ButtonLink from 'components/Button/ButtonLink'

type NextLessonProps = {
  difficultyLevel: string,
  nextLesson?: {
    lessonId: string,
    lessonTitle: string,
  },
}

const Wrap = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
`

const NextLesson = ({ difficultyLevel, nextLesson }: NextLessonProps) => (
  <GridSection>
    <Wrap>
      {nextLesson ? (
        <>
          <strong>Nästa lektion: {nextLesson.lessonTitle}</strong>
          <ButtonLink to={`/lesson/${nextLesson.lessonId}`}>
            Fortsätt &gt;
          </ButtonLink>
        </>
      ) : (
        <>
          Bra jobbat, du är färdig! Tillbaka till översiktssidan
          <ButtonLink to={`/overview/${difficultyLevel}`}>Tillbaka</ButtonLink>
        </>
      )}
    </Wrap>
  </GridSection>
)

export default NextLesson
