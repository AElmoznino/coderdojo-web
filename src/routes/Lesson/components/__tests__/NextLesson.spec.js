import React from 'react'
import { shallow } from 'enzyme'
import NextLesson from '../NextLesson'

describe('components/NextLesson', () => {
  let component

  beforeEach(() => {
    component = shallow(<NextLesson difficultyLevel="nybÃ¶rjare" />)
  })

  it('renders NextLesson when there is no next lesson', () => {
    expect(component).toMatchSnapshot()
  })

  it('renders NextLesson where there is a next lesson', () => {
    component.setProps({
      nextLesson: {
        lessonId: 'test-id',
        lessonTitle: 'test-title',
      },
    })

    expect(component).toMatchSnapshot()
  })
})
