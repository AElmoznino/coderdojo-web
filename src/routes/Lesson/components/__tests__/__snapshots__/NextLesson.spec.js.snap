// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NextLesson renders NextLesson when there is no next lesson 1`] = `
<styled.div>
  <styled.div>
    Bra jobbat, du är färdig! Tillbaka till översiktssidan
    <Styled(Link)
      to="/overview/nybörjare"
    >
      Tillbaka
    </Styled(Link)>
  </styled.div>
</styled.div>
`;

exports[`components/NextLesson renders NextLesson where there is a next lesson 1`] = `
<styled.div>
  <styled.div>
    <strong>
      Nästa lektion: 
      test-title
    </strong>
    <Styled(Link)
      to="/lesson/test-id"
    >
      Fortsätt &gt;
    </Styled(Link)>
  </styled.div>
</styled.div>
`;
